benchmarks/high_volume_writes_test.go

```diff
@@ -50,69 +57,55 @@ var _ = Describe("Scenario: High Volume Writes", func() {
-		b.RecordValue("All ingester successful GRPC push QPS", qps)
-
-		// Record BoltDB Shipper Writes QPS
-		qps, _ = metricsClient.RequestBoltDBShipperWritesQPS(job, defaultRange)
-		b.RecordValue("All boltdb shipper successful writes QPS", qps)
-
-		// Record p99 loki_request_duration_seconds_bucket
-		p99, err = metricsClient.RequestDurationOkGrpcPushP99(job, defaultRange)
-		Expect(err).Should(Succeed(), "Failed to read p99 for all ingester GRPC push with status code 2xx")
-		b.RecordValue("All ingester successful GRPC push p99", p99)
-
-		// Record p50 loki_request_duration_seconds_bucket
-		p50, err = metricsClient.RequestDurationOkGrpcPushP50(job, defaultRange)
-		Expect(err).Should(Succeed(), "Failed to read p50 for all ingester GRPC push with status code 2xx")
-		b.RecordValue("All ingester successful GRPC push p50", p50)
-
-		// Record avg from loki_request_duration_seconds_sum / loki_request_duration_seconds_count
-		avg, err = metricsClient.RequestDurationOkGrpcPushAvg(job, defaultRange)
-		Expect(err).Should(Succeed(), "Failed to read average for all ingester GRPC push with status code 2xx")
-		b.RecordValue("All ingester successful GRPC push avg", avg)
-
-		mu.Lock()
-		defer mu.Unlock()
-		totalSamples -= 1
-
-	}, benchCfg.Scenarios.HighVolumeWrites.Samples.Total)
+	for _, configuration := range benchCfg.Scenarios.HighVolumeWrites.Configurations {
+		c := configuration // Make a local copy to avoid the "Using the variable on range scope `verb` in function literal"
+		Measure("should result in measurements - configuration: "+c.Description, func(b Benchmarker) {
+			defaultRange := scenarioCfg.Configurations[configurationCount].Samples.Range
+
+			// Collect measurements for distributors
+			job := benchCfg.Metrics.DistributorJob()
+			err := metricsClient.Measure(b, metricsClient.RequestWritesQPS, "2xx push QPS", job, c.Description, defaultRange )
+			Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v",err))
+			err = metricsClient.Measure(b, metricsClient.RequestDurationOkPushP99, "2xx push p99", job, c.Description, defaultRange )
+			Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v",err))
+			err = metricsClient.Measure(b, metricsClient.RequestDurationOkPushP50, "2xx push p50", job, c.Description, defaultRange )
+			Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v",err))
+			err = metricsClient.Measure(b, metricsClient.RequestDurationOkPushAvg, "2xx push avg", job, c.Description, defaultRange )
+			Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v",err))
+
+			// Collect measurements for Ingesters
+			job = benchCfg.Metrics.IngesterJob()
+			caJob := benchCfg.Metrics.CaIngesterJob()
+			err = metricsClient.Measure(b, metricsClient.ProcessCPU, "Processes CPU", job, c.Description, defaultRange )
+			Expect(err).Should(Succeed(), fmt.Sprintf("Failed - %v",err))
+			if benchCfg.Logger.Namespace != "default" { 	//skip none-relevant metrics for development environment (cadvisor not deployed)
```

Comment:
Can we make this an explicit flag in the config?, e.g. `benchCfg.metrics.enableCadvisorMetrics`

Reply:
done